resources/agent.rb

```diff
@@ -24,18 +24,23 @@
 property :installer_url, String, default: lazy { "http://www.appveyor.com/downloads/deployment-agent/#{version}/AppveyorDeploymentAgent.msi" }
 property :install_path, String, default: lazy { 'C:\\Program Files (x86)\\AppVeyor\\DeploymentAgent\\Appveyor.DeploymentAgent.Service.exe' }
 
-action :create do
+default_action :install
+
+action :install do
   include_recipe 'windows'
 
-  package 'AppveyorDeploymentAgent' do
+  windows_package 'AppveyorDeploymentAgent' do
```

Comment:
```
  package 'AppveyorDeploymentAgent' do
    source installer_url
    installer_type :msi
    options "/quiet /qn /norestart /log install.log ENVIRONMENT_ACCESS_KEY=#{access_key}"
  end
```
This was code used to install the package but installer_type is not the property of package but it is for windows_package. Instead of it and try, I change the resource to windows_package. We can try package resource removing the installer_type property. It should be intelligent enough to understand the type of installation.